wagtail/admin/views/generic/chooser.py

```diff
@@ -221,6 +221,28 @@ def columns(self) -> list[Column]:
             columns.append(LocaleColumn(sort_key=None))
         return columns
 
+    def _get_edit_url_for_object(self, obj):
+        """
+        Get the edit URL for a given object instance.
+        """
+        return AdminURLFinder(user=self.request.user).get_edit_url(obj)
```

Comment:
Does this have sensible behaviour when the user does not have edit permission over the item being shown?

Reply:
> Yes, I tried with a user which don't have edit permissoins (moderator permissions group), and it works as expected (read only view).

This doesn't seem to be the case in my tests - if `AdminURLFinder.get_edit_url` returns None (which always happens for a user without edit permission - I see nothing in there that would cause it to fall back on the inspect view), no link is generated at all and so the item cannot be chosen.

Given that the inspect view [is an optional feature](https://docs.wagtail.org/en/latest/topics/snippets/customizing.html#inspect-view) that won't be present for all model types, I think the correct fix here is something akin to #9209 - using a button instead of a link, so that there is no native right-click / middle-click behaviour provided by the browser at all.